feature-libs/product-configurator/rulebased/components/composition/configurator-attribute-composition.directive.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Directive,
+  Injector,
+  Input,
+  OnInit,
+  ViewContainerRef,
+} from '@angular/core';
+import { ConfiguratorAttributeCompositionConfig } from './configurator-attribute-composition.config';
+import { ConfiguratorAttributeCompositionContext } from './configurator-attribute-composition.model';
+
+@Directive({
+  selector: '[cxConfiguratorAttributeComponent]',
+})
+export class ConfiguratorAttributeCompositionDirective implements OnInit {
+  @Input('cxConfiguratorAttributeComponent')
+  context: ConfiguratorAttributeCompositionContext;
```

Comment:
Just for this directive to be bullet-proof, let's discuss what should happen when the value of the context `@Input()` changes over time?
Shouldn't we reset the `vcr` and do the `renderComponent` again?

Reply:
@Platonn TBH I don't understand when this would happen, because on every new submission of a runtime configuration, e.g. on changing an attribute value, new instances of the directive are created.
(Is that somewhat surprising to you?)